core/controllers/classroom.py

```diff
@@ -359,3 +377,137 @@ def get(self, classroom_url_fragment: str) -> None:
         self.render_json({
             'classroom_id': classroom.classroom_id
         })
+
+
+class NewClassroomDataHandlerNormalizedPayloadDict(TypedDict):
+    """Dict representation of NewClassroomHandler's
+    normalized_payload dictionary.
+    """
+
+    name: str
+    url_fragment: str
+
+
+class NewClassroomHandler(
+    base.BaseHandler[
+        NewClassroomDataHandlerNormalizedPayloadDict, Dict[str, str]
+    ]
+):
+    """Creates a new classroom."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'name': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'has_length_at_most',
+                        'max_value': constants.MAX_CHARS_IN_CLASSROOM_NAME
+                    }, {
+                        'id': 'is_nonempty',
+                    }]
+                }
+            },
+            'url_fragment': constants.SCHEMA_FOR_CLASSROOM_URL_FRAGMENTS,
+        }
+    }
+
+    @acl_decorators.open_access
+    def post(self) -> None:
```

Comment:
Argh, sorry for not noticing this in the first place. There's already a PUT /classroom handler, can't we reuse that? If the passed id doesn't exist, create a new classroom.

If it's not difficult to reuse class/function/handler, it's better to do so, because more entities - more mental burden and more things one needs to comprehend

Reply:
Yeah sorry I forgot to tell. yes earlier even in the proposal I mentioned to use the same handler for both operations since we were having same properties in both the cases but now while creating we will just need name and url fragment I asked Sean about this. So I guess it makes sense to just have a separate enpoint for it just like we have one for creating topic.